File: lib/models/syslog_message.dart
import 'package:equatable/equatable.dart';

class SyslogMessage extends Equatable {
  final DateTime timestamp;
  final String host;
  final String severity;
  final int severityCode;
  final String facility;
  final int facilityCode;
  final String message;
  final String rawMessage;

  const SyslogMessage({
    required this.timestamp,
    required this.host,
    required this.severity,
    required this.severityCode,
    required this.facility,
    required this.facilityCode,
    required this.message,
    required this.rawMessage,
  });

  factory SyslogMessage.fromJson(Map<String, dynamic> json) {
    return SyslogMessage(
      timestamp: DateTime.parse(json['timestamp'] as String),
      host: json['host'] as String? ?? '',
      severity: json['severity'] as String? ?? 'Unknown',
      severityCode: json['severity_code'] as int? ?? 0,
      facility: json['facility'] as String? ?? 'Unknown',
      facilityCode: json['facility_code'] as int? ?? 0,
      message: json['message'] as String? ?? '',
      rawMessage: json['raw_message'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'timestamp': timestamp.toIso8601String(),
    'host': host,
    'severity': severity,
    'severity_code': severityCode,
    'facility': facility,
    'facility_code': facilityCode,
    'message': message,
    'raw_message': rawMessage,
  };

  @override
  List<Object?> get props => [timestamp, host, severity, severityCode, facility, facilityCode, message, rawMessage];
}

enum Severity {
  emergency(0, 'Emergency', 0xFFFF5252),
  alert(1, 'Alert', 0xFFFF5252),
  critical(2, 'Critical', 0xFFFF5252),
  error(3, 'Error', 0xFFFF9800),
  warning(4, 'Warning', 0xFFFFEB3B),
  notice(5, 'Notice', 0xFF2196F3),
  info(6, 'Info', 0xFF9E9E9E),
  debug(7, 'Debug', 0xFF4CAF50);

  final int code;
  final String label;
  final int color;

  const Severity(this.code, this.label, this.color);

  static Severity fromCode(int code) {
    return Severity.values.firstWhere(
      (s) => s.code == code,
      orElse: () => Severity.info,
    );
  }
}

File: go/message/syslog_message.go
package message

import "time"

type SyslogMessage struct {
	Timestamp    time.Time `json:"timestamp"`
	Host         string    `json:"host"`
	Severity     string    `json:"severity"`
	SeverityCode int       `json:"severity_code"`
	Facility     string    `json:"facility"`
	FacilityCode int       `json:"facility_code"`
	Message      string    `json:"message"`
	RawMessage   string    `json:"raw_message"`
}

// Severity constants (syslog RFC 5424)
const (
	SeverityEmergency = 0
	SeverityAlert     = 1
	SeverityCritical  = 2
	SeverityError     = 3
	SeverityWarning   = 4
	SeverityNotice    = 5
	SeverityInfo      = 6
	SeverityDebug     = 7
)

// Facility constants
const (
	FacilityKernel     = 0
	FacilityUser       = 1
	FacilityMail       = 2
	FacilityDaemon     = 3
	FacilityAuth       = 4
	FacilitySyslog     = 5
	FacilityLpr        = 6
	FacilityNews       = 7
	FacilityUUCP       = 8
	FacilityCron       = 9
	FacilityAuthPriv   = 10
	FacilityFTP        = 11
	FacilityNTP        = 12
	FacilityLogAudit   = 13
	FacilityLogAlert   = 14
	FacilityLocal0     = 16
	FacilityLocal1     = 17
	FacilityLocal2     = 18
	FacilityLocal3     = 19
	FacilityLocal4     = 20
	FacilityLocal5     = 21
	FacilityLocal6     = 22
	FacilityLocal7     = 23
)

func SeverityToString(code int) string {
	switch code {
	case SeverityEmergency:
		return "Emergency"
	case SeverityAlert:
		return "Alert"
	case SeverityCritical:
		return "Critical"
	case SeverityError:
		return "Error"
	case SeverityWarning:
		return "Warning"
	case SeverityNotice:
		return "Notice"
	case SeverityInfo:
		return "Info"
	case SeverityDebug:
		return "Debug"
	default:
		return "Unknown"
	}
}

func FacilityToString(code int) string {
	switch code {
	case FacilityKernel:
		return "Kernel"
	case FacilityUser:
		return "User"
	case FacilityMail:
		return "Mail"
	case FacilityDaemon:
		return "Daemon"
	case FacilityAuth:
		return "Auth"
	case FacilitySyslog:
		return "Syslog"
	case FacilityLpr:
		return "Lpr"
	case FacilityNews:
		return "News"
	case FacilityUUCP:
		return "UUCP"
	case FacilityCron:
		return "Cron"
	case FacilityAuthPriv:
		return "AuthPriv"
	case FacilityFTP:
		return "FTP"
	case FacilityNTP:
		return "NTP"
	case FacilityLogAudit:
		return "LogAudit"
	case FacilityLogAlert:
		return "LogAlert"
	case FacilityLocal0:
		return "Local0"
	case FacilityLocal1:
		return "Local1"
	case FacilityLocal2:
		return "Local2"
	case FacilityLocal3:
		return "Local3"
	case FacilityLocal4:
		return "Local4"
	case FacilityLocal5:
		return "Local5"
	case FacilityLocal6:
		return "Local6"
	case FacilityLocal7:
		return "Local7"
	default:
		return "Unknown"
	}
}
